### PR TITLE
Refactor/code samples language config

### PR DESCRIPTION
**Pull request by JulianVallee**
## Summary

This PR started as a set of adjustments I needed for my own usage, but it turned out to be a broader improvement that other users may find useful as well. It consolidates code sample language configuration into a single `LanguageConfig` type and extends support to all languages available in Shiki and `@scalar/snippetz`. Previously, only a small set of hardcoded languages was supported. With this change, users can add any supported language, define multiple clients for the same language (e.g. TypeScript with `fetch` vs `axios`), and provide fully custom languages with their own generator.

Disclaimer for transparency: AI was obviously used *for assistance, codegen and debugging*, but manually vetted and refactored everything that I touched, so this PR should contain minimal to no vibe slop

### Consolidated LanguageConfig

Previously, language configuration was spread across multiple places: a `langs` array for Shiki highlighter names, a separate `availableLanguages` array for the UI, and hardcoded `languagesMap`/`clientsMap` inside `generateCodeSample.ts` to map identifiers to `@scalar/snippetz` targets. Adding a new language meant modifying internal source code.

Now, everything lives in a single `LanguageConfig` object:

```ts
interface LanguageConfig {
  lang: string         // Arbitrary identifier for referencing
  label: string        // Name shown in the language selector
  target?: string      // @scalar/snippetz generation target
  client?: string      // HTTP client library for generated code
  highlighter?: string // Shiki language for syntax highlighting
  icon?: string        // Icon identifier for vitepress-plugin-group-icons
}
```

Users can filter defaults, add languages, define variants, or provide custom generators, all through `useTheme()` configuration.

### Shiki: support for all bundled languages

Shiki now loads only core highlighters (`json`, `xml`, `markdown`) on init, as these are needed for response body highlighting. All other grammars are loaded via dynamic imports through `ensureLanguage()`, meaning any of Shiki's [bundled languages](https://shiki.style/languages) can be used for code sample syntax highlighting.

> Note: this does not implement true lazy loading (all code sample grammars still load at page render since all tabs are mounted). Per-tab lazy loading can be added in a follow-up PR, the plumbing (`ensureLanguage()` + the `active` prop) is already in place.

### @scalar/snippetz: support for all plugins

Code generation now uses `target` and `client` directly from `LanguageConfig` instead of mapping through hardcoded lookup tables. Any [snippetz plugin](https://github.com/scalar/scalar/tree/main/packages/snippetz/src/plugins) can be used by setting the corresponding `target`/`client` values on a language config.

### Tests and documentation

- Extended and added tests covering custom languages and edge cases
- Updated [code samples documentation](https://github.com/enzonotario/vitepress-openapi/blob/main/docs/customizations/code-samples.md) with sections for default languages, filtering, adding languages, multiple variants, and custom generators

## Screenshot

*Support for new languages*
image-1

*Support for multi-client samples (@scalar/snippets does not support TS, using `target: 'js'` here)*
image

### `generator` signature changed

The generator function now receives `(langConfig: LanguageConfig, request: OARequest)` instead of `(lang: string, request)`.

**Before:**
```ts
generator: async (lang, request) => {
  if (lang === 'bruno') return generateBru(request)
  return generateCodeSample(lang, request)
}
```

**After:**
```ts
generator: async (langConfig, request) => {
  if (langConfig.lang === 'bruno') return generateBru(request)
  return generateCodeSample(langConfig, request)
}
```

### `langs` config removed

The `codeSamples.langs` config no longer exists. If you were using it to control which Shiki highlighters to load, set the `highlighter` field on each entry in `availableLanguages` instead.

### `getCodeSamplesLangs()` removed

Use `getCodeSamplesAvailableLanguages()` instead, which returns `LanguageConfig[]`. To get just the language identifiers:

```ts
useTheme().getCodeSamplesAvailableLanguages().map(l => l.lang)
```

### `codeSamples` prop removed from `OACodeSamples`

The component now sources its data internally via `useTheme()`. Remove any `:code-samples` bindings from parent components.

### CODE REVIEW

File: src/composables/useShiki.ts [hunk 1]
```diff
@@ -54,10 +77,62 @@ export function useShiki() {
...
+   * Ensure a language is loaded. Loads it on-demand if not already available.
+   * Returns true if the language is ready, false if it couldn't be loaded.
+   */
+  async function ensureLanguage(lang: string): Promise<boolean> {
+    // Make sure shiki is initialized first
+    await init()
+
+    if (!shiki) {
+      return false
+    }
```
[the remark this answers is not in the document]
Comment (JulianVallee) [reply src/composables/useShiki.ts:95]: Will do this manually, IIRC the suggestion is incomplete and breaks some tests that currently expect promise rejections

File: src/lib/codeSamples/generateCodeSample.ts [hunk 2]
```diff
@@ -1,32 +1,33 @@
...
+  }
+
+  if (!client) {
+    console.error(`Language "${lang}" has missing property "client" in availableLanguages`)
+    return ''
+  }
+
   try {
-    return snippetz().print(languagesMap[lang], clientsMap[lang], harRequest) ?? ''
+    return snippetz().print(target as any, client as any, harRequest) ?? ''
```
[the remark this answers is not in the document]
Comment (JulianVallee) [reply src/lib/codeSamples/generateCodeSample.ts:30]: Type `any` was a deliberate decision here, `LanguageConfig` props are regular strings and typescript was throwing a fit when I cast them to the proposed correct types. Should be fine here though, we have additional checks in place before hitting this line

@enzonotario what do you think, can we live with that?

File: src/composables/useTheme.ts [hunk 3]
```diff
@@ -187,16 +186,24 @@ export interface LinksPrefixesConfig {
...
   label: string
-  highlighter: string
+  /** Target language for @scalar/snippetz code generation (e.g., 'js', 'shell', 'python') */
+  target?: string
+  /** HTTP client for the target language (e.g., 'fetch', 'axios', 'curl') */
+  client?: string
+  /** Icon identifier for vitepress-plugin-group-icons */
   icon?: string
+  /** Shiki language for syntax highlighting (defaults to 'plain') */
+  highlighter?: string
```
[the remark this answers is not in the document]
Comment (JulianVallee) [reply src/composables/useTheme.ts:201]: Will resolve this manually, getting error "Applying suggestions on deleted lines is currently not supported."

File: src/composables/useShiki.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (JulianVallee) [reply src/composables/useShiki.ts:95]: Well, I guess... good bot? 😄 

Change was actually fine and covered by "allows retry after initialization failure" in useShiki.test.ts

File: src/components/Sample/OACodeSamples.vue [hunk 4]
```diff
@@ -22,42 +26,59 @@ const themeConfig = useTheme()
...
+    }),
   )
+
+  if (!activeSampleKey.value || !samples.value.some(s => s.key === activeSampleKey.value)) {
+    const defaultLang = themeConfig.getCodeSamplesDefaultLang()
+    const defaultSample = samples.value.find(s => s.lang === defaultLang) || samples.value[0]
+    if (defaultSample) {
+      activeSampleKey.value = defaultSample.key
+    }
+  }
```
[the remark this answers is not in the document]
Comment (JulianVallee) [reply src/components/Sample/OACodeSamples.vue]: - Added proper guards
- Removed `immediate` to prevent watcher running twice
- Improved UX, now with explicit hint instead of hard-to-debug empty string

Unbenannt

**Comment by enzonotario** [comment]
Awesome @JulianVallee ! It's now released in [v0.1.14](https://github.com/enzonotario/vitepress-openapi/releases/tag/v0.1.14).

I added a preload because E2E were failing due to the time it takes to load each language: #318 .

Many thanks!
